source/builders.txt

```diff
@@ -0,0 +1,334 @@
+    
+   $filter = Query::<logical operator>(
+       Query::query(<field name>: Query::<factory method>(<parameters>)),
+       Query::query(<field name>: Query::<factory method>(<parameters>)),
+       ...
+   );
+
+To learn more, see :manual:`Logical Query Operators
+</reference/operator/query-logical/>` in the {+mdb-server+} manual.
+
+The following sections provide examples that use builders to create
+filter definitions for different operations.
+
+Retrieve Example
+~~~~~~~~~~~~~~~~
+
+This example performs the following actions:
+
+- Uses the ``Query::eq()`` factory method to match documents in which
+  the ``feature_type`` field value is ``'Wrecks - Visible'``
+- Uses the ``Query::near()`` factory method to match documents in which
+  the ``coordinates`` location field is within ``10000`` meters of the
+  specified coordinates
+- Calls the :phpmethod:`MongoDB\Collection::find()`
+  method to retrieve the matching documents
+- Prints the matching documents
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-find
+      :end-before: end-find
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: json
+      :visible: false
+
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9137115,9.3390503],...}
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9357223,9.3340302],...}
+      {"_id":...,"feature_type":"Wrecks - Visible","coordinates":[-79.9081268,9.3547792],...}
+      // Results truncated
+
+To learn more about find operations, see the :ref:`php-retrieve` guide.
+
+Delete Example
+~~~~~~~~~~~~~~
+
+This example performs the following actions:
+
+- Uses the ``Query::or()`` factory method to match documents that
+  satisfy the either of the following query clauses:
+  
+  - Clause that uses the ``Query::regex()`` factory method to check if
+    the ``feature_type`` field value contains the string ``'nondangerous'``
+  
+  - Clause that uses the ``Query::gt()`` factory method to check if the
+    ``depth`` field value is greater than ``10.0``
+
+- Calls the :phpmethod:`MongoDB\Collection::deleteOne()`
+  method to delete the first matching document
+
+- Prints the number of deleted documents
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-deleteone
+      :end-before: end-deleteone
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Deleted documents: 1
+
+To learn more about delete operations, see the :ref:`php-write-delete`
+guide.
+
+.. _php-builders-update:
+
+Define an Update Document
+-------------------------
+
+You can use the ``Stage::set()`` factory method when creating update
+documents. Update documents describe the updates to make to target
+documents. To learn more about updating documents, see the
+:ref:`php-write-update` guide.
+
+.. important:: Only $set Operator Supported in Update Documents
```

Comment:
That's confusing. 
Update supports an [aggregation pipeline](https://www.mongodb.com/docs/manual/tutorial/update-documents-with-aggregation-pipeline/), in that case most of the stages can be used.

Or by default that's [update operators](https://www.mongodb.com/docs/manual/reference/operator/update/) that are not supported by the builder yet. They are not stages even if the syntax for `$set` is similar.

Reply:
Yes, you can remove it. We will try to add this as a new feature: [PHPLIB-1638](https://jira.mongodb.org/browse/PHPLIB-1638)